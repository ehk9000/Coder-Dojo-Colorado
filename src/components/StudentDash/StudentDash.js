import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import requests from '../../utils/requests/requests';


export function StudentDash(props) {
  const [pastEvents, setAttendedEvents] = useState({});
  const { user } = props;

  const getEventsAttended = async () => {
    const attendedEvents = await requests.getEventsAttended();
    setAttendedEvents(attendedEvents.me.eventsAttended);
    console.log(attendedEvents.me.eventsAttended);
  };

  useEffect(() => {
    getEventsAttended(user.id);
  }, [user.id]);

  const generateEvents = events => events.map(event => <p>{event.name}</p>);

  return (
    <section className="StudentDashContainer">
      <div className="StudentDashHeader">
        <h2>
          Welcome, {user.name}
        </h2>
        
      </div>
      <form>
        <input 
          type="text" 
          placeholder="Enter Event Code"
        />
        <button type="submit">Submit</button>
      </form>
      {pastEvents.length > 0 && generateEvents(pastEvents)}
    </section>
  );
}

export const mapStateToProps = ({ user, isFetching, error }) => ({
  user,
  isFetching,
  error
});

export default connect(mapStateToProps, null)(StudentDash);

StudentDash.propTypes = {
  user: PropTypes.object,
  isFetching: PropTypes.bool,
  error: PropTypes.string
};

StudentDash.defaultProps = {
  user: {},
  isFetching: false,
  error: ''
};
