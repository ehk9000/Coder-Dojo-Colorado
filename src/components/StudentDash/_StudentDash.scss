.StudentDashContainer {
  width:80%;
  margin: auto;
  height:100vh;

  .studentDashHeader {
    background-color: #724695;
    width: 100%;
    padding: 30px 0;
    display: flex;
    align-items: center;
    color:#FFFFFF;
  }

  h2 {
    font-weight: 400;
    margin: 0 40px;
  }

  .eventWrapper {
    @include flex-column;
    margin: 5% auto;
    width: 600px;
    min-width: 300px;
    padding: 40px;
    background-color: #FFFFFF;
  }

  .eventCode {
    @include form-input-style;
    width:300px;
  }

  .eventCode::placeholder {
    text-align: center;
  }

  button {
    @include form-submit($primary-color);
    background-color: $footer-color;
    border-color:  $footer-color;
    margin-left: 60px;
    width: 125px;
  }
  .pastEvents {
    @include flex-column;
    max-height: 500px;
    overflow: scroll;
  }
}

