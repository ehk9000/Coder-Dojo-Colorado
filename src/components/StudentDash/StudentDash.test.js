import React from 'react';
import { shallow } from 'enzyme';
import { StudentDash, mapStateToProps } from './StudentDash';



describe('StudentDash', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<StudentDash />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should MSTP', () => {
    const state = {
      error: ''
    };

    const { error } = mapStateToProps(state);

    expect(error).toEqual(state.error);
  });
  it('should MSTP with user', () => {
    const state = {
      user: {
        id: 1,
        username: 'ehk9000'
      }
    };

    const { username } = mapStateToProps(state);

    expect(username).toEqual(state.username);
  });
  it('should MSTP with isFetching', () => {
    const state = {
      isFetching: true
    };

    const { isFetching } = mapStateToProps(state);

    expect(isFetching).toEqual(state.isFetching);
  });
});
