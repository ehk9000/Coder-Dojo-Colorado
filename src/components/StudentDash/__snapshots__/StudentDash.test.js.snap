// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDash should match snapshot 1`] = `
<section
  className="StudentDashContainer"
>
  <div
    className="studentDashHeader"
  >
    <h2>
      Welcome, 
    </h2>
  </div>
  <h3>
    Current Event
  </h3>
  <div
    className="eventWrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="eventCode"
        onChange={[Function]}
        placeholder="Enter Event Code"
        type="text"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <section
    className="pastEvents"
  />
</section>
`;
